kedro/io/core.py

```diff
@@ -57,6 +57,7 @@
 HTTP_PROTOCOLS = ("http", "https")
 PROTOCOL_DELIMITER = "://"
 CLOUD_PROTOCOLS = ("s3", "gcs", "gs", "adl", "abfs")
+SMB_PROTOCOLS = "smb"
```

Comment:
```suggestion
SMB_PROTOCOLS = ("smb", )
```

Reply:
Also, will there be other protocols in addition to samba which should have paths parsed this way? i.e. should this variable be something more generic than `SMB_PROTOCOLS`? Like `NETWORK_PROTOCOLS` or something?

Reply:
I used to work in the past with FTP protocols but not anymore but I can imagine that they will need to parsed in a similar way:
https://filesystem-spec.readthedocs.io/en/latest/_modules/fsspec/implementations/ftp.html
There is a whole bunch of other protocols such as git file system but I dont know if they are that useful to Kedro:
https://filesystem-spec.readthedocs.io/en/latest/_modules/index.html

Reply:
> 
> 
> This looks great to me, thank you for your contribution @lucasjamar! I'm not very familiar with these protocols so will try and get some people who are more familiar to take a look.
> 
> Please could you explain to a dummy why the whole `username:password@myhost:8080` get passed as a path here when it doesn't for s3 etc.?

I dont know why this doesnt happen for S3. This would be a question for the author of the original code. In general, I dont see what the _parse_filepath is for since it splits the path into a protocol and path and then feeds that into fsspec. This is something that fsspec handles itself but it additionally parses the host, username, port, etc so in the case of Kedro's _parse_filepath function, stuff like the host, port, username, etc get lost. Its even noted in the _parse_filepath code that it has the same functionality as `fsspec.utils.infer_storage_options` so I dont see why that function is not used instead.
To be clear, my current implementation is not absolutely necessary. There is a workaround.  `urlsplit` considers the first string after `smb://` as the network location. Kedro then excludes that from the path (why, Im guessing because of other filesystem storages such as http or hdfs?). So my workaround has been to provide a dummy netloc and feed stuff like the host, port, username and password via fs_args. Here is an example:
```
raw_iris:
  type: pandas.CSVDataSet
  filepath: smb://dummy_url/folder/iris.csv
  fs_args:
    host: <hostname>
    port: 8080
```

However it feels clumsy and it does beg the point of why the netlwork location was removed in the first place.

Thanks for your help @AntonyMilneQB !

Reply:
> 
> 
> I believe @AntonyMilneQB 's code suggestion is still a valid one that should be addressed.

Sorry @lorenabalan ! I had integrated the parentheses, but then I applied black which automatically removed them. I just fixed it again

Reply:
The problem is that you're missing a trailing comma - it should be:
```
SMB_PROTOCOLS = ("smb", )
```
Notice the `,` before `)`.

What you currently have is this:
```
SMB_PROTOCOLS = ("smb")
```
... which is interpreted by Python as if the parentheses aren't there at all (and so black strips them out). This sets `SMB_PROTOCOLS` to the string `smb`. Including the comma sets it to a tuple with one element (the string `smb`). This means that looking up membership (`if protocol in SMB_PROTOCOLS`) will work. 🙂

Reply:
I'm going to try to look your questions about fsspec and the way we're handling different protocols tomorrow btw - sorry for the delay, it's been a busy couple of days!